psydac/ddm/tests/test_partition.py

```diff
@@ -48,6 +48,30 @@ def test_partition_1d_general( mpi_size ):
     with pytest.raises( Exception ):
         dims, blocksizes = compute_dims( mpi_size, [n1,], [p1,] )
 
+#==============================================================================
+@pytest.mark.parametrize( 'mpi_size', [1,2,5,10] )
+@pytest.mark.parametrize( 'mask', [[True, False], [False, True]] )
+@pytest.mark.parametrize( 'npts', [[64, 64], [58, 64], [64, 31]] )
+
+def test_partition_2d_dims_mask( mpi_size, npts, mask ):
+
+    print(f'\nmpi_size={mpi_size}:')    
+    # General partition: blocks are not all identical but closer to a cube
+    dims, blocksizes = compute_dims( mpi_size, npts, [3,3] )   
+    print(f'general:'.ljust(30), dims, blocksizes) 
+    
+    # Mask dimensions
+    dims, blocksizes = compute_dims( mpi_size, npts, [3,3], dims_mask=mask )   
+    print(f'dims_mask {mask}:'.ljust(30), dims, blocksizes) 
```

Comment:
Although we still have print statements in some of our tests, Pytest is not happy about it. Here you should pass the flag `verbose = False` to the new functions, and only print to terminal if this flag is set to `True`. Then you can pass `verbose = True` when you run the script manually (bottom of the file).

Or you can simply remove the print statements, of course.

Reply:
Ok I will add `verbose=False`. What do you mean by Pytest does not like it? Can there be problems?

Reply:
By default Pytest collects the standard output and only shows it if there is an error or warning. This means that if a test fails the error message will be cluttered with the output from all the other tests (which do not fail). This is far from ideal.